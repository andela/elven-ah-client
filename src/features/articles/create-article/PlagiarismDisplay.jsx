import React from 'react';
import PropTypes from 'prop-types';

const PlagiarismDisplay = ({ scanResults }) => (
  <React.Fragment>
    <br /><br />
    <span className="invalid-feedback">The content of your article was dicovered below.</span>
    <br />
    <ul className="list-group">
      {scanResults.map(({ title, url, totalMatchedPercents }) => (
        <li className="list-group-item d-flex justify-content-between align-items-center" key={`${title}${url}`}>
          <a href={url} rel="noopener noreferrer" target="_blank">
            {title}
          </a>
          <span className="badge badge-primary badge-pill">{totalMatchedPercents}%</span>
        </li>
      ))}
    </ul>
    <br />
    <span className="invalid-feedback">Please ensure that you resolve them or attribute all relevant sources.</span>
  </React.Fragment>
);

PlagiarismDisplay.propTypes = {
  scanResults: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export default PlagiarismDisplay;
