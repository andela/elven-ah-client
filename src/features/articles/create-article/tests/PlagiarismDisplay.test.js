import React from 'react';
import { shallow } from 'enzyme';
import PlagiarismDisplay from '../PlagiarismDisplay';

const scanResults = [
  {
    title: 'Plagiarism Article',
    url: 'https://medium.com/test-user/plagiarised-article-mi9i4f',
    totalMatchedPercents: 34,
  },
  {
    title: 'Plagiarism Article',
    url: 'https://medium.com/test-user/plagiarised-article-mi9i4f',
    totalMatchedPercents: 34,
  },
  {
    title: 'Plagiarism Article',
    url: 'https://medium.com/test-user/plagiarised-article-mi9i4f',
    totalMatchedPercents: 34,
  },
];

describe('Plagiarism Display component', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<PlagiarismDisplay scanResults={scanResults} />);
    expect(wrapper.find('ul')).toHaveLength(1);
    expect(wrapper.find('li')).toHaveLength(3);
    expect(wrapper).toMatchSnapshot();
  });
});
