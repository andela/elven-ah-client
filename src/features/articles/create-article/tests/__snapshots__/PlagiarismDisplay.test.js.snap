// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plagiarism Display component should render without errors 1`] = `
<Fragment>
  <br />
  <br />
  <span
    className="invalid-feedback"
  >
    The content of your article was dicovered below.
  </span>
  <br />
  <ul
    className="list-group"
  >
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      key="Plagiarism Articlehttps://medium.com/test-user/plagiarised-article-mi9i4f"
    >
      <a
        href="https://medium.com/test-user/plagiarised-article-mi9i4f"
        rel="noopener noreferrer"
        target="_blank"
      >
        Plagiarism Article
      </a>
      <span
        className="badge badge-primary badge-pill"
      >
        34
        %
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      key="Plagiarism Articlehttps://medium.com/test-user/plagiarised-article-mi9i4f"
    >
      <a
        href="https://medium.com/test-user/plagiarised-article-mi9i4f"
        rel="noopener noreferrer"
        target="_blank"
      >
        Plagiarism Article
      </a>
      <span
        className="badge badge-primary badge-pill"
      >
        34
        %
      </span>
    </li>
    <li
      className="list-group-item d-flex justify-content-between align-items-center"
      key="Plagiarism Articlehttps://medium.com/test-user/plagiarised-article-mi9i4f"
    >
      <a
        href="https://medium.com/test-user/plagiarised-article-mi9i4f"
        rel="noopener noreferrer"
        target="_blank"
      >
        Plagiarism Article
      </a>
      <span
        className="badge badge-primary badge-pill"
      >
        34
        %
      </span>
    </li>
  </ul>
  <br />
  <span
    className="invalid-feedback"
  >
    Please ensure that you resolve them or attribute all relevant sources.
  </span>
</Fragment>
`;
